File: rest/recording/list-get-example-4/list-get-example-4.8.x.java
// Install the Java helper library from twilio.com/docs/java/install
import com.twilio.Twilio;
import com.twilio.base.ResourceSet;
import com.twilio.rest.api.v2010.account.Recording;

import java.time.ZonedDateTime;

public class Example {
    // Get your Account SID and Auth Token from https://twilio.com/console
    // To set up environment variables, see http://twil.io/secure
    public static final String ACCOUNT_SID = System.getenv("TWILIO_ACCOUNT_SID");
    public static final String AUTH_TOKEN = System.getenv("TWILIO_AUTH_TOKEN");

    public static void main(String[] args) {
        Twilio.init(ACCOUNT_SID, AUTH_TOKEN);

        ZonedDateTime lower = ZonedDateTime.of(2016, 10, 12, 0, 0, 0, 0, ZoneId.of("UTC"));
        ZondedDateTime upper = ZonedDateTime.of(2016, 10, 15, 0, 0, 0, 0, ZoneId.of("UTC"));

        ResourceSet<Recording> recordings = Recording.reader()
                .setDateCreatedAfter(lower)
                .setDateCreatedBefore(upper)
                .read();

        // Loop over recordings and print out a property for each one.
        for (Recording recording : recordings) {
            System.out.println(recording.getDuration());
        }
    }
}

File: rest/notification/list-get-example-4/list-get-example-4.8.x.py
# Download the Python helper library from twilio.com/docs/python/install
import os
from twilio.rest import Client
from datetime import date

# Your Account Sid and Auth Token from twilio.com/user/account
# To set up environmental variables, see http://twil.io/secure
account_sid = os.environ['TWILIO_ACCOUNT_SID']
auth_token = os.environ['TWILIO_AUTH_TOKEN']
client = Client(account_sid, auth_token)

# A list of notification objects with the properties described above
notifications = client.notifications \
                      .list(message_date_after=date(2009, 7, 6),
                            message_date_before=date(2009, 7, 8),
                            log="1")

for notification in notifications:
    print(notification.request_url)
